The new release of Pytest Asyncio (0.25.1 and 0.25.2) makes a number of asyncio tests fail

After Pytest-asyncio 0.25.1 our canary tests started to fail with "RuntimeError: There is no current event loop in thread 'MainThread'." This happens randomly - a lot of times in Python 3.12 non-db tests, but sometimes they succeed there, and fail in other Python versions. It happens with xdist with xdist n=1 and without xdist alike. 

Example failures:

* https://github.com/apache/airflow/actions/runs/12583221724/job/35070671929#step:6:904
* https://github.com/apache/airflow/actions/runs/12579028168/job/35066133378#step:6:904

Stacktrace: 

```python
_ ERROR at setup of TestRunPipelineJobTrigger.test_run_yields_success_event_on_successful_pipeline_state[4] _
[gw2] linux -- Python 3.11.11 /usr/local/bin/python
src/python/grpcio/grpc/_cython/_cygrpc/aio/common.pyx.pxi:184: in grpc._cython.cygrpc.get_working_loop
    ???
E   RuntimeError: no running event loop

During handling of the above exception, another exception occurred:
providers/tests/google/cloud/triggers/test_vertex_ai.py:133: in pipeline_service_async_client
    return PipelineServiceAsyncClient(
/usr/local/lib/python3.11/site-packages/google/cloud/aiplatform_v1/services/pipeline_service/async_client.py:304: in __init__
    self._client = PipelineServiceClient(
/usr/local/lib/python3.11/site-packages/google/cloud/aiplatform_v1/services/pipeline_service/client.py:[889](https://github.com/apache/airflow/actions/runs/12583221724/job/35070671929#step:6:906): in __init__
    self._transport = transport_init(
/usr/local/lib/python3.11/site-packages/google/cloud/aiplatform_v1/services/pipeline_service/transports/grpc_asyncio.py:310: in __init__
    self._grpc_channel = channel_init(
/usr/local/lib/python3.11/site-packages/google/cloud/aiplatform_v1/services/pipeline_service/transports/grpc_asyncio.py:177: in create_channel
    return grpc_helpers_async.create_channel(
/usr/local/lib/python3.11/site-packages/google/api_core/grpc_helpers_async.py:290: in create_channel
    return aio.secure_channel(
/usr/local/lib/python3.11/site-packages/grpc/aio/_channel.py:621: in secure_channel
    return Channel(
/usr/local/lib/python3.11/site-packages/grpc/aio/_channel.py:367: in __init__
    self._loop = cygrpc.get_working_loop()
src/python/grpcio/grpc/_cython/_cygrpc/aio/common.pyx.pxi:186: in grpc._cython.cygrpc.get_working_loop
    ???
src/python/grpcio/grpc/_cython/_cygrpc/aio/common.pyx.pxi:190: in grpc._cython.cygrpc.get_working_loop
    ???
/usr/local/lib/python3.11/asyncio/events.py:681: in get_event_loop
    raise RuntimeError('There is no current event loop in thread %r.'
E   RuntimeError: There is no current event loop in thread 'MainThread'.
_ ERROR at setup of TestRunPipelineJobTrigger.test_run_yields_success_event_on_successful_pipeline_state[8] _
[gw2] linux -- Python 3.11.11 /usr/local/bin/python
src/python/grpcio/grpc/_cython/_cygrpc/aio/common.pyx.pxi:184: in grpc._cython.cygrpc.get_working_loop
    ???
E   RuntimeError: no running event loop
```

Failing tests:

```
ERROR providers/tests/google/cloud/triggers/test_vertex_ai.py::TestRunPipelineJobTrigger::test_run_yields_success_event_on_successful_pipeline_state[4] - RuntimeError: There is no current event loop in thread 'MainThread'.
ERROR providers/tests/google/cloud/triggers/test_vertex_ai.py::TestRunPipelineJobTrigger::test_run_yields_success_event_on_successful_pipeline_state[8] - RuntimeError: There is no current event loop in thread 'MainThread'.
ERROR providers/tests/google/cloud/triggers/test_vertex_ai.py::TestRunPipelineJobTrigger::test_run_yields_error_event_on_failed_pipeline_state[5] - RuntimeError: There is no current event loop in thread 'MainThread'.
ERROR providers/tests/google/cloud/triggers/test_vertex_ai.py::TestRunPipelineJobTrigger::test_run_yields_error_event_on_failed_pipeline_state[7] - RuntimeError: There is no current event loop in thread 'MainThread'.
ERROR providers/tests/google/cloud/triggers/test_vertex_ai.py::TestCustomTrainingJobTrigger::test_run_yields_success_event_on_successful_pipeline_state[4] - RuntimeError: There is no current event loop in thread 'MainThread'.
ERROR providers/tests/google/cloud/triggers/test_vertex_ai.py::TestCustomTrainingJobTrigger::test_run_yields_success_event_on_successful_pipeline_state[8] - RuntimeError: There is no current event loop in thread 'MainThread'.
ERROR providers/tests/google/cloud/triggers/test_vertex_ai.py::TestCustomTrainingJobTrigger::test_run_yields_error_event_on_failed_pipeline_state[5] - RuntimeError: There is no current event loop in thread 'MainThread'.
ERROR providers/tests/google/cloud/triggers/test_vertex_ai.py::TestCustomTrainingJobTrigger::test_run_yields_error_event_on_failed_pipeline_state[7] - RuntimeError: There is no current event loop in thread 'MainThread'.
ERROR providers/tests/google/cloud/triggers/test_vertex_ai.py::TestCustomContainerTrainingJobTrigger::test_run_yields_success_event_on_successful_pipeline_state[4] - RuntimeError: There is no current event loop in thread 'MainThread'.
ERROR providers/tests/google/cloud/triggers/test_vertex_ai.py::TestCustomContainerTrainingJobTrigger::test_run_yields_success_event_on_successful_pipeline_state[8] - RuntimeError: There is no current event loop in thread 'MainThread'.
ERROR providers/tests/google/cloud/triggers/test_vertex_ai.py::TestCustomContainerTrainingJobTrigger::test_run_yields_error_event_on_failed_pipeline_state[5] - RuntimeError: There is no current event loop in thread 'MainThread'.
ERROR providers/tests/google/cloud/triggers/test_vertex_ai.py::TestCustomContainerTrainingJobTrigger::test_run_yields_error_event_on_failed_pipeline_state[7] - RuntimeError: There is no current event loop in thread 'MainThread'.
ERROR providers/tests/google/cloud/triggers/test_vertex_ai.py::TestCustomPythonPackageTrainingJobTrigger::test_run_yields_success_event_on_successful_pipeline_state[4] - RuntimeError: There is no current event loop in thread 'MainThread'.
ERROR providers/tests/google/cloud/triggers/test_vertex_ai.py::TestCustomPythonPackageTrainingJobTrigger::test_run_yields_success_event_on_successful_pipeline_state[8] - RuntimeError: There is no current event loop in thread 'MainThread'.
ERROR providers/tests/google/cloud/triggers/test_vertex_ai.py::TestCustomPythonPackageTrainingJobTrigger::test_run_yields_error_event_on_failed_pipeline_state[5] - RuntimeError: There is no current event loop in thread 'MainThread'.
ERROR providers/tests/google/cloud/triggers/test_vertex_ai.py::TestCustomPythonPackageTrainingJobTrigger::test_run_yields_error_event_on_failed_pipeline_state[7] - RuntimeError: There is no current event loop in thread 'MainThread'.
```


Also https://github.com/apache/airflow/actions/runs/12666006582/job/35297011421#step:6:4025

```
==================================== ERRORS ====================================
__ ERROR at teardown of TestAirbyteSyncTrigger.test_airbyte_run_sync_trigger ___
/usr/local/lib/python3.11/site-packages/pytest_asyncio/plugin.py:1151: in event_loop
    with _temporary_event_loop_policy(new_loop_policy), _provide_event_loop() as loop:
/usr/local/lib/python3.11/contextlib.py:144: in __exit__
    next(self.gen)
/usr/local/lib/python3.11/site-packages/pytest_asyncio/plugin.py:1168: in _provide_event_loop
    loop.run_until_complete(loop.shutdown_asyncgens())
/usr/local/lib/python3.11/asyncio/base_events.py:652: in run_until_complete
    raise RuntimeError('Event loop stopped before Future completed.')
E   RuntimeError: Event loop stopped before Future completed.
----------------------------- Captured stdout call -----------------------------
[2025-01-08T07:44:52.728+0000] {base.py:66} INFO - Retrieving connection 'airbyte_default'
------------------------------ Captured log call -------------------------------
INFO     airflow.hooks.base:base.py:66 Retrieving connection 'airbyte_default'
____________ ERROR at teardown of TestTrigger.test_trigger_run_bad _____________
/usr/local/lib/python3.11/site-packages/pytest_asyncio/plugin.py:1151: in event_loop
    with _temporary_event_loop_policy(new_loop_policy), _provide_event_loop() as loop:
/usr/local/lib/python3.11/contextlib.py:144: in __exit__
    next(self.gen)
/usr/local/lib/python3.11/site-packages/pytest_asyncio/plugin.py:1168: in _provide_event_loop
    loop.run_until_complete(loop.shutdown_asyncgens())
/usr/local/lib/python3.11/asyncio/base_events.py:652: in run_until_complete
    raise RuntimeError('Event loop stopped before Future completed.')
E   RuntimeError: Event loop stopped before Future completed.
______ ERROR at teardown of TestWasbBlobSensorTrigger.test_running[True] _______
/usr/local/lib/python3.11/site-packages/pytest_asyncio/plugin.py:1151: in event_loop
    with _temporary_event_loop_policy(new_loop_policy), _provide_event_loop() as loop:
/usr/local/lib/python3.11/contextlib.py:144: in __exit__
    next(self.gen)
/usr/local/lib/python3.11/site-packages/pytest_asyncio/plugin.py:1168: in _provide_event_loop
    loop.run_until_complete(loop.shutdown_asyncgens())
/usr/local/lib/python3.11/asyncio/base_events.py:652: in run_until_complete
    raise RuntimeError('Event loop stopped before Future completed.')
E   RuntimeError: Event loop stopped before Future completed.
______ ERROR at teardown of TestWasbBlobSensorTrigger.test_running[False] ______
```

--- a/devel-common/pyproject.toml
+++ b/devel-common/pyproject.toml
@@ -124,7 +124,7 @@ dependencies = [
     # General pytest devel tools
     # We need to adjust all our tests to work with "proper" handling of the async loops in pytest-asyncio
     # Implemented in Pytest-asyncio 0.25.1 and 0.25.2. See: https://github.com/apache/airflow/issues/45355
-    "pytest-asyncio>=0.24.0,<0.25.1",
+    "pytest-asyncio>=1.1.0",
     "pytest-cov>=4.1.0",
     "pytest-custom-exit-code>=0.3.0",
     "pytest-icdiff>=0.9",

--- a/devel-common/pyproject.toml
+++ b/devel-common/pyproject.toml
@@ -122,8 +122,6 @@ dependencies = [
 ]
 "pytest" = [
     # General pytest devel tools
-    # We need to adjust all our tests to work with "proper" handling of the async loops in pytest-asyncio
-    # Implemented in Pytest-asyncio 0.25.1 and 0.25.2. See: https://github.com/apache/airflow/issues/45355
     "pytest-asyncio>=1.1.0",
     "pytest-cov>=4.1.0",
     "pytest-custom-exit-code>=0.3.0",

--- a/providers/cncf/kubernetes/tests/unit/cncf/kubernetes/hooks/test_kubernetes.py
+++ b/providers/cncf/kubernetes/tests/unit/cncf/kubernetes/hooks/test_kubernetes.py
@@ -27,6 +27,7 @@ from unittest.mock import MagicMock, PropertyMock, patch
 import anyio
 import kubernetes
 import pytest
+import pytest_asyncio
 import yaml
 from kubernetes.client import V1Deployment, V1DeploymentStatus
 from kubernetes.client.rest import ApiException
@@ -847,8 +848,8 @@ class TestAsyncKubernetesHook:
         f.set_result(return_value)
         return f
 
-    @pytest.fixture
-    def kube_config_loader(self):
+    @pytest_asyncio.fixture
+    async def kube_config_loader(self):
         with mock.patch(self.KUBE_LOADER_CONFIG) as kube_config_loader:
             kube_config_loader.return_value.load_and_set.return_value = self.mock_await_result(None)
             yield kube_config_loader

--- a/providers/cncf/kubernetes/tests/unit/cncf/kubernetes/operators/test_spark_kubernetes.py
+++ b/providers/cncf/kubernetes/tests/unit/cncf/kubernetes/operators/test_spark_kubernetes.py
@@ -28,6 +28,7 @@ from uuid import uuid4
 
 import pendulum
 import pytest
+import pytest_asyncio
 import yaml
 from kubernetes.client import models as k8s
 
@@ -45,8 +46,8 @@ from tests_common.test_utils.version_compat import AIRFLOW_V_3_0_PLUS
 POD_MANAGER_CLASS = "airflow.providers.cncf.kubernetes.utils.pod_manager.PodManager"
 
 
-@pytest.fixture(autouse=True, scope="module")
-def patch_pod_manager_methods():
+@pytest_asyncio.fixture(autouse=True, scope="function")
+async def patch_pod_manager_methods():
     # Patch watch_pod_events
     patch_watch_pod_events = mock.patch(f"{POD_MANAGER_CLASS}.watch_pod_events", new_callable=mock.AsyncMock)
     mock_watch_pod_events = patch_watch_pod_events.start()

--- a/providers/google/tests/unit/google/cloud/hooks/test_kubernetes_engine.py
+++ b/providers/google/tests/unit/google/cloud/hooks/test_kubernetes_engine.py
@@ -24,6 +24,7 @@ from unittest import mock
 
 import kubernetes.client
 import pytest
+import pytest_asyncio
 from google.cloud.container_v1 import ClusterManagerAsyncClient
 from google.cloud.container_v1.types import Cluster
 
@@ -543,8 +544,8 @@ class TestGKEKubernetesAsyncHook:
         assert "Test string #2" in caplog.text
 
 
-@pytest.fixture
-def async_gke_hook():
+@pytest_asyncio.fixture
+async def async_gke_hook():
     return GKEAsyncHook(
         gcp_conn_id=GCP_CONN_ID,
         location=GKE_ZONE,
@@ -552,8 +553,8 @@ def async_gke_hook():
     )
 
 
-@pytest.fixture
-def mock_async_gke_cluster_client():
+@pytest_asyncio.fixture
+async def mock_async_gke_cluster_client():
     f = Future()
     f.set_result(None)
     client = mock.MagicMock(spec=ClusterManagerAsyncClient)

--- a/providers/google/tests/unit/google/cloud/hooks/vertex_ai/test_custom_job.py
+++ b/providers/google/tests/unit/google/cloud/hooks/vertex_ai/test_custom_job.py
@@ -21,6 +21,7 @@ import asyncio
 from unittest import mock
 
 import pytest
+import pytest_asyncio
 
 # For no Pydantic environment, we need to skip the tests
 pytest.importorskip("google.cloud.aiplatform_v1")
@@ -58,23 +59,23 @@ BASE_STRING = "airflow.providers.google.common.hooks.base_google.{}"
 CUSTOM_JOB_STRING = "airflow.providers.google.cloud.hooks.vertex_ai.custom_job.{}"
 
 
-@pytest.fixture
-def test_async_hook():
+@pytest_asyncio.fixture
+async def test_async_hook():
     return CustomJobAsyncHook(
         gcp_conn_id=TEST_GCP_CONN_ID,
         impersonation_chain=TEST_IMPERSONATION_CHAIN,
     )
 
 
-@pytest.fixture
-def pipeline_service_async_client():
+@pytest_asyncio.fixture
+async def pipeline_service_async_client():
     return PipelineServiceAsyncClient(
         credentials=mock.MagicMock(),
     )
 
 
-@pytest.fixture
-def job_service_async_client():
+@pytest_asyncio.fixture
+async def job_service_async_client():
     return JobServiceAsyncClient(
         credentials=mock.MagicMock(),
     )

--- a/providers/google/tests/unit/google/cloud/triggers/test_vertex_ai.py
+++ b/providers/google/tests/unit/google/cloud/triggers/test_vertex_ai.py
@@ -20,6 +20,7 @@ import asyncio
 from unittest import mock
 
 import pytest
+import pytest_asyncio
 
 # For no Pydantic environment, we need to skip the tests
 pytest.importorskip("google.cloud.aiplatform_v1")
@@ -112,24 +113,24 @@ def custom_training_job_trigger():
     )
 
 
-@pytest.fixture
-def custom_job_async_hook():
+@pytest_asyncio.fixture
+async def custom_job_async_hook():
     return CustomJobAsyncHook(
         gcp_conn_id=TEST_CONN_ID,
         impersonation_chain=TEST_IMPERSONATION_CHAIN,
     )
 
 
-@pytest.fixture
-def pipeline_job_async_hook():
+@pytest_asyncio.fixture
+async def pipeline_job_async_hook():
     return PipelineJobAsyncHook(
         gcp_conn_id=TEST_CONN_ID,
         impersonation_chain=TEST_IMPERSONATION_CHAIN,
     )
 
 
-@pytest.fixture
-def pipeline_service_async_client():
+@pytest_asyncio.fixture
+async def pipeline_service_async_client():
     return PipelineServiceAsyncClient(
         credentials=mock.MagicMock(),
     )

